src/common/dnnl_thread.cpp

```diff
@@ -0,0 +1,101 @@
+#include <functional>
+
+#include "dnnl_thread.hpp"
+
+#if defined(DNNL_ENABLE_ITT_TASKS)
+#include "common/ittnotify.hpp"
+#endif
+
+#if DNNL_CPU_THREADING_RUNTIME == DNNL_RUNTIME_THREADPOOL
+#include "counting_barrier.hpp"
+#endif
+
+namespace dnnl {
+namespace impl {
+
+void parallel(int nthr, const std::function<void(int, int)> &f) {
+    nthr = adjust_num_threads(nthr, INT64_MAX);
```

Comment:
as far as I remember, if function is not inline, compiler cannot optimize code properly and performance is worse.

Reply:
yes, it can be so
But implementation via separate cpp and header file was the same in the previous openvino release.
So, we won't regression in comparison with the previous release

Reply:
no, I was not able to reproduce it with oneDNN tests (benchdnn / gtests)
But when I enable ITT only for CPU plugin and not oneDNN, the issue is not reproduced

Reply:
Which platforms do you see segfaults?

Are you sure that this is a pure compiler issue?

We moved threading back to header file because otherwise it can't be used from our validation since an original copy of symbol will be from the library and no-cpu configuration validation can't be sped up. We unlikely move it again to cpp file. Some performance regressions were caused by lack of vectorization in plain copies because compiler can't optimize it indeed. If you see such places, please let us know.